### ISSUE TITLE
image_picker: Cancel button not visible in gallery, if camera was accessed first

### ISSUE THREAD

**Issue by santervo** [flutter/flutter#21922]
## Steps to Reproduce

1. Open image picker with image source camera (must be done first after application has started)
2. Open image picker with image source gallery

Result: Cancel and Back buttons are not shown on gallery view

## Logs

```
[+8007 ms] [VERBOSE-2:engine.cc(158)] Could not prepare to run the isolate.
[   +1 ms] [VERBOSE-2:engine.cc(117)] Engine not prepare and launch isolate.
[   +1 ms] [VERBOSE-2:FlutterViewController.mm(437)] Could not launch engine with configuration.
```

```
Analyzing flutter_image_picker_test...
No issues found! (ran in 1.6s)
```

```
[✓] Flutter (Channel master, v0.8.2-pre.25, on Mac OS X 10.13.4 17E202, locale en-FI)
    • Flutter version 0.8.2-pre.25 at /Users/santtu/Development/flutter/sdk
    • Framework revision bd1e5a361b (6 days ago), 2018-09-10 10:27:01 +0200
    • Engine revision 1770f88548
    • Dart version 2.1.0-dev.4.0.flutter-cd9a42239f

[!] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/santtu/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • ANDROID_HOME = /Users/santtu/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)
    ✗ Android license status unknown.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.4.1, Build version 9F2000
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.2

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    • Dart plugin version 162.2924
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[!] VS Code (version 1.27.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected devices (1 available)
    • iPhone  • xxx • ios • iOS 11.2.2

! Doctor found issues in 2 categories.
```

## To reproduce

Here's main file for reproducting the issue. Plugin image_picker must be installed in the app also.

```dart
import 'package:flutter/material.dart';
import 'package:image_picker/image_picker.dart';

void main() => runApp(new App());

class App extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
        title: 'Image Picker test',
        theme: new ThemeData(
          primarySwatch: Colors.blue,
        ),
        home: new ImagePickerTest());
  }
}

class ImagePickerTest extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return SafeArea(
      child: Column(
        children: <Widget>[
          new RaisedButton(
            child: new Text("Camera (click 1st)"),
            onPressed: () {
              ImagePicker.pickImage(source: ImageSource.camera);
            },
          ),
          new RaisedButton(
            child: new Text("Gallery (click 2nd)"),
            onPressed: () {
              ImagePicker.pickImage(source: ImageSource.gallery);
            },
          ),
        ],
      ),
    );
  }
}
```

**Comment by Sukendh** [comment]
posted the same issue [here](https://stackoverflow.com/questions/37999095/uinavigationbar-disappears-after-uiimagepickercontroller-is-popped). Thanks

**Comment by lordgreg** [comment]
Same issue here.

**Comment by zoechi** [comment]
#23903 contains some additional details about the cause

**Comment by cyanglaz** [comment]
Should be fixed in 0.5.4+1

**Comment by Gustavoo44** [comment]
I fixed it using try catch as follow:

try {
      await picker
          .pickImage(
              source: ImageSource.camera,
              maxHeight: 700,
              maxWidth: 700,
              imageQuality: 100)
          .then((value) async {
        pickedFile = value;
        result = true;
        await imageSaved(File(pickedFile!.path));
      });
    } catch (e) {
      print(e);
    }
